postgres/datadog_checks/postgres/statement_samples.py

```diff
@@ -271,9 +281,10 @@ def _close_db_pool(self):
     def _collect_statement_samples(self):
         self._rate_limiter.sleep()
         start_time = time.time()
+        max_query_size = self._get_query_max_text_size()
```

Comment:
Since `track_activity_query_size` can only be changed after restarting the database, there's no reason to keep repeating the query here as this happens once per second by default. 

We should cache the result after the first read, only re-reading it out after the database connection has been recreated (if the database is restarted then they may have changed the parameter).

Reply:
That was my intention initially and I don't remember why I didn't do it that way. It does make more sense to do it only once when connecting (or reconnecting). I'll change.

Reply:
I just pushed this change. It's a bit awkward because the new connection is established in `_get_db` so I'm checking there if the dbname is equal to `_config.dbname` to only load/reload the query_max_text_size then. Let me know what you think.